frontend/src/pages/main/transactions/components/generatePdf/GeneratePdf.tsx

```diff
@@ -0,0 +1,202 @@
+    const loggedUser = useAppSelector(selectLoggedUser);
+    const dispatch = useAppDispatch();
+
+    const updateDays = (selectedMonth: number, selectedYear: number, periodType: PeriodType) => {
+        const daysInMonth = new Date(selectedYear || 0, (selectedMonth || 0) + 1, 0).getDate();
+        if (periodType === PeriodType.START) {
+            setStartDays(Array.from({length: daysInMonth}, (_, i) => i + 1));
+        } else {
+            setEndDays(Array.from({length: daysInMonth}, (_, i) => i + 1));
+        }
+    }
+
+    const updateReportUser = () => {
+        setReport({...report, userId: loggedUser.id});
+    }
+
+    const handleDaysChange = (event: SelectChangeEvent<number>, periodType: PeriodType) => {
+        const day = event.target.value as number;
+        if (periodType === PeriodType.START) {
+            setStartPeriod({...startPeriod, day});
+            setReport({...report, periodStart: new Date(startPeriod.year, startPeriod.month, day)});
+        } else {
+            setEndPeriod({...endPeriod, day});
+            setReport({...report, periodEnd: new Date(endPeriod.year, endPeriod.month, day)});
+        }
+    };
+
+    const handleMonthChange = (event: SelectChangeEvent<number>, periodType: PeriodType) => {
+        const month = event.target.value as number;
+        if (periodType === PeriodType.START) {
+            setStartPeriod({...startPeriod, month});
+            setReport({
+                ...report,
+                periodStart: new Date(startPeriod.year, month, startPeriod.day)
+            })
+        } else {
+            setEndPeriod({...endPeriod, month});
+            setReport({
+                ...report,
+                periodEnd: new Date(endPeriod.year, month, endPeriod.day)
+            })
+        }
+        updateDays(month, startPeriod.year, periodType);
+    };
+
+    const handleYearChange = (event: SelectChangeEvent<number>, periodType: PeriodType) => {
+        const year = event.target.value as number;
+        if (periodType === PeriodType.START) {
+            setStartPeriod({...startPeriod, year});
+            setReport({
+                ...report,
+                periodStart: new Date(year, startPeriod.month, startPeriod.day)
+            })
+        } else {
+            setEndPeriod({...endPeriod, year});
+            setReport({
+                ...report,
+                periodEnd: new Date(year, endPeriod.month, endPeriod.day)
+            })
+        }
+        updateDays(startPeriod.month, year, periodType);
+    };
+
+    const handleGenerate = async () => {
+        if (report.periodStart.getTime() > report.periodEnd.getTime()) {
+            dispatch(showNotification({show: true, type: NotificationType.ERROR, message: Notifications.PERIOD_TIME}));
+            return;
+        }
+        dispatch(showNotification({
+            show: true,
+            type: NotificationType.SUCCESS,
+            message: Notifications.SEND_GENERATE_PDF
+        }));
```

Comment:
You have to show successful notification , after request is successful.
Just move it to the next line after ` await sendReport(report);`